### PR TITLE
[FIX TRAVIS] Make the new node on a graph not depend on the old node.

### PR DESCRIPTION
**Pull request by nouiz**
This fix the master of travis.

the problem was triggered by gh-4128. But it was already there before. It just that now we don't clean up the problem in fast_compile.

### CODE REVIEW

**Comment by abergeron** [comment]
Ok for me.

File: theano/tensor/opt.py [hunk 1]
```diff
@@ -3753,8 +3753,21 @@ def local_useless_switch(node):
...
+                for idx, (b1, b2), in enumerate(zip(out.type.broadcastable,
+                                                    node.outputs[0].type.broadcastable)):
+                    if b1 == b2:
+                        shps.append(out.shape[idx])
+                    elif not node.inputs[1].type.broadcastable[idx]:
+                        shps.append(node.inputs[1].shape[idx])
+                    else:
+                        shps.append(node.inputs[2].shape[idx])
+                out = T.alloc(out, *shps)
+                return False
```
Comment (lamblin) [inline theano/tensor/opt.py]: I think that `return False` is what is currently making `test_local_useless_switch` fail.

Comment (nouiz) [inline theano/tensor/opt.py]: I ran locally those tests with this change and this fix it. I updated this PR.

**Comment by abergeron** [comment]
Again, I'm ok when travis passes.

**Pull request merged**